aten/src/ATen/Context.cpp

```diff
@@ -205,32 +261,69 @@ void Context::setBenchmarkLimitCuDNN(int b) {
 }
 
 bool Context::allowTF32CuBLAS() const {
-  return float32_matmul_precision != at::Float32MatmulPrecision::HIGHEST;
+  bool legacy_allow_tf32 = float32_matmul_precision != at::Float32MatmulPrecision::HIGHEST;
+  bool allow_tf32 = float32Precision("cuda", "matmul") == "tf32" ? true : false;
+  TORCH_CHECK(
+      legacy_allow_tf32 == allow_tf32,
+      "You are checking whether allow_tf32 is enabled for cublas matmul. ",
+      "From Pytorch 2.4 we introduced an new set of API to set the TF32 status for cublas matmul. ",
+      "Current status indicate that you have used mix of legacy API and new API to set the TF32 status for cublas matmul. ",
+      "We suggest to only use the new API to set the TF32 status, you can check the usage at ",
+      "https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst#tensorfloat-32-tf32-on-ampere-and-later-devices");
+  return allow_tf32;
 }
 
 void Context::setAllowTF32CuBLAS(bool b) {
   float32_matmul_precision = b ? at::Float32MatmulPrecision::HIGH : at::Float32MatmulPrecision::HIGHEST;
+  setFloat32Precision(b ? "tf32" : "default", "cuda", "matmul");
 }
 
 Float32MatmulPrecision Context::float32MatmulPrecision() const {
+  bool invalid = float32Precision("cuda", "matmul") == "tf32" &&
+      float32_matmul_precision == at::Float32MatmulPrecision::HIGHEST;
+  invalid = invalid ||
+      (float32Precision("mkldnn", "matmul") == "bf16" &&
+       float32_matmul_precision != at::Float32MatmulPrecision::MEDIUM);
+  TORCH_CHECK(
+      !invalid,
+      "From Pytorch 2.4 we introduced an new set of API to set the TF32 status "
+      "for fp32 matmul precision in a more fine-grained way. ",
+      "Current status indicate that you have used mix of legacy API and new API",
+      "You have used the new API to turn the status into an un-represented state for old API. ",
+      "And now you are using old API to query the status."
+      "We suggest to only use the new API, you can check the usage at ",
+      "https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst#tensorfloat-32-tf32-on-ampere-and-later-devices");
   return float32_matmul_precision;
```

Comment:
Same as above, is there a reason we can't just convert and only maintain one set of flags?

Reply:
Same as above